src/main/java/com/mraof/minestuck/entity/dialogue/Trigger.java

```diff
@@ -220,6 +221,62 @@ public void triggerEffect(LivingEntity entity, ServerPlayer player)
 		}
 	}
 	
+	Codec<EquipmentSlot> EQUIPMENT_CODEC = Codec.STRING.xmap(
+			EquipmentSlot::byName,
```

Comment:
There should probably be some error-catching here because `EquipmentSlot.byName()` might throw an `IllegalArgumentException`. The codec should produce a `DataResult.error()` instead of throwing.

Reply:
Im not sure how to do that, sorry

Reply:
```java
static DataResult<EquipmentSlot> parseSlot(String slotName)
	{
		try
		{
			return DataResult.success(EquipmentSlot.byName(slotName));
		} catch(IllegalArgumentException e)
		{
			return DataResult.error(() -> "Exception trying to parse an EquipmentSLot");
		}
	}
	
	Codec<EquipmentSlot> EQUIPMENT_CODEC = Codec.STRING.comapFlatMap(
			Trigger::parseSlot,
			EquipmentSlot::getName
	);
```

Reply:
Yeah, that should work! I'd like to suggest a slightly different error message:
`return DataResult.error(() -> "Not a valid name for an EquipmentSlot: " + slotName);`